lib/transaction.py

```diff
@@ -755,8 +756,24 @@ def sign(self, keypairs):
                     secexp = pkey.secret
                     private_key = MySigningKey.from_secret_exponent(secexp, curve = SECP256k1)
                     public_key = private_key.get_verifying_key()
-                    sig = private_key.sign_digest_deterministic(pre_hash, hashfunc=hashlib.sha256, sigencode = ecdsa.util.sigencode_der)
-                    assert public_key.verify_digest(sig, pre_hash, sigdecode = ecdsa.util.sigdecode_der)
+   
+                    use_Schnorr_Sigs = False
+                    Schnorr_Fail = False
+
+                    # Turn on for DEBUG:
+                    # use_Schnorr_Sigs = True
+
+                    if use_Schnorr_Sigs:
+                        schnorr_pkey=bytes.fromhex(private_key.to_string().hex())
+                        schnorr_msg = bytes.fromhex(pre_hash.hex())
+                        sig = sign_schnorr(schnorr_msg,schnorr_pkey)
+                        if len(sig) < 64:
```

Comment:
This is just more nasty spaghetti.  We can't riddle the codebase with if everywhere. This approach is not good.